### PR TITLE
feature: prepare web for deployment

### PR DESCRIPTION
**Pull request by geclos**
Prepares web to be deployed to fly.io

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -34,15 +34,13 @@ services:
...
       dockerfile: apps/web/docker/Dockerfile
-    # This enables to have this service for testing production build
-    # It will not run when doing docker compose up
-    profiles: [building]
-    volumes:
-      - ./packages/env:/app/packages/env
-      - ./packages/core:/app/packages/core
-      - ./packages/jobs:/app/packages/jobs
-      - ./packages/web-ui:/app/packages/web-ui
-      - ./apps/web:/app/apps/web
```
Comment (andresgutgon) [inline docker-compose.yml:45]: I find this helpful

File: docker-compose.yml [hunk 2]
```diff
@@ -34,15 +34,13 @@ services:
...
+    ports:
+      - 3000:3000
+    environment:
+      - NODE_ENV=production
+      - DATABASE_URL=postgres://latitude:secret@db:5432/latitude_production
+      - REDIS_PORT=6379
+      - REDIS_HOST=redis
+      - REDIS_PASSWORD=secret
```
Comment (andresgutgon) [inline docker-compose.yml:44]: Why is this necessary?

File: apps/web/docker/Dockerfile [hunk 3]
```diff
@@ -1,4 +1,15 @@
-FROM latitudedata/base AS base
+FROM node:20-alpine AS base
```
Comment (andresgutgon) [inline apps/web/docker/Dockerfile:1]: With only these changes the image builds. I find it better to have an install of dependencies for build and then another stage for `--prod` dependencies. Take into account if you change this

File: docker-compose.yml (hunk 1 above)
Comment (geclos) [reply docker-compose.yml:45]: why? are u using the dockerfile for development?

File: docker-compose.yml (hunk 2 above)
Comment (geclos) [reply docker-compose.yml:44]: not needed just using it to test the production image locally

File: apps/web/src/data-access/users.ts [hunk 4]
```diff
@@ -34,7 +34,7 @@ export async function getUserFromCredentials({
   if (!user) return notFound()
 
   const validPassword = await verifyPassword(password, user.encryptedPassword)
-  if (!validPassword) notFound()
+  if (!validPassword) return notFound()
```
Comment (csansoon) [inline apps/web/src/data-access/users.ts:37]: nice catch

**Review by csansoon — APPROVED** [review]
